### PR TITLE
Feature/weather infrastructure

### PR DESCRIPTION
**Pull request by afontani**
## Pull Request Description

**Companion PR:** resstock-estimation [#95](https://github.com/NREL/resstock-estimation/pull/95)

This PR generalizes the weather data available for a given AMY or TMY analysis. ResStock currently uses 941 static weather stations. The stations are used based on the Location.tsv.

The issue with the current approach is that if the data from a particular weather station 1) has errors or does not exist for the year of interest and/or 2) is not consistent with corresponding TMY stations the weather file name in options_lookup.tsv or the Location.tsv needs to change to simulate different years or run a TMY analysis. 

In an effort to simply replace the weather zip file based on the best available data for the given analysis (AMY or TMY), the weather files have been named based on their county GEOID in options_lookup.tsv.

**Example:**
```
County | AL, Autauga County | ResidentialLocation | dst_start_date=April 7 | dst_end_date=October 26 | weather_directory=../../../../weather | weather_file_name=G0100010.epw
```
The new weather zip files are named `BuildStock_<year>.zip` where year is for example 2019, 2018, ... or TMY3. In these files there exists a mapping file (example: `map_FIPs_TMY3s.csv`) that states the weather station that is used for a given county and other meta data. In this approach, the Counties that correspond to a given weather station is flexible and not static.

### New TMY3 weather data
TMY3 weather data that correspond to the new structure can be downloaded from this link: [https://data.nrel.gov/submissions/156](https://data.nrel.gov/submissions/156)

## Checklist

Not all may apply:

- [ ] Unit tests have been added or updated
- [x] All rake tasks have been run, and pass
- [ ] Documentation has been modified appropriately
- [x] Any new options are added to `project_testing`
- [ ] `project_testing` runs without any failures
- [x] No unexpected regression test changes
- [x] All tests are passing (green) on circleci
- [x] The [changelog](https://github.com/NREL/resstock/blob/master/CHANGELOG.md) has been updated appropriately
- [x] This branch is up-to-date with develop

For more information on how to perform these checklist items, see the documentation's [Advanced Tutorial](https://resstock.readthedocs.io/en/latest/advanced_tutorial/index.html).

### CODE REVIEW

**Comment by afontani** [comment]
@joseph-robertson: The weather data is available to download (https://data.nrel.gov/submissions/156). I updated the national.yml file to reflect the correct download path. If this comes back green, I think this is about ready.

**Review by joseph-robertson — CHANGES_REQUESTED** [review]
This needs to be changed as well, right? https://github.com/NREL/resstock/blob/feature/weather_infrastructure/project_testing/testing.yml#L6

File: project_testing/housing_characteristics/County.tsv [hunk 1]
```diff
@@ -1,2 +1,3 @@
...
...V, Monroe County	Option=WV, Morgan County	Option=WV, Nicholas County	Option=WV, Ohio County	Option=WV, Pendleton County	Option=WV, Pleasants County	Option=WV, Pocahontas County	Option=WV, Preston County	Option=WV, Putnam County	Option=WV, Raleigh County	Option=WV, Randolph County	Option=WV, Ritchie County	Option=WV, Roane County	Option=WV, Summers County	Option=WV, Taylor County	Option=WV, Tucker County	Option=WV, Tyler County	Option=WV, Upshur County	Option=WV, Wayne County	Option=WV, Webster County	Option=WV, Wetzel County	Option=WV, Wirt County	Option=WV, Wood County	Option=WV, Wyoming County	Option=WY, Albany County	Option=WY, Big Horn County	Option=WY, Campbell County	Option=WY, Carbon County	Option=WY, Converse County	Option=WY, Crook County	Option=WY, Fremont County	Option=WY, Goshen County	Option=WY, Hot Springs County	Option=WY, Johnson County	Option=WY, Laramie County	Option=WY, Lincoln County	Option=WY, Natrona County	Option=WY, Niobrara County	Option=WY, Park County	Option=WY, Platte County	Option=WY, Sheridan County	Option=WY, Sublette County	Option=WY, Sweetwater County	Option=WY, Teton County	Option=WY, Uinta County	Option=WY, Washakie County	Option=WY, Weston County
```
Comment (joseph-robertson) [inline project_testing/housing_characteristics/County.tsv:1]: I think you should choose up to 6 counties as options. Corresponding epw files will need to be added to `resources/measures/HPXMLtoOpenStudio/weather/project_testing` (which are used in unit testing).

Comment (afontani) [reply project_testing/housing_characteristics/County.tsv:1]: Oh. You are right. I think this commit is going to fail the tests.

**Review by joseph-robertson — APPROVED** [review]

**Pull request merged**